lib/phoenix/live_dashboard/pages/ecto_stats_page.ex

```diff
@@ -3,63 +3,194 @@ defmodule Phoenix.LiveDashboard.EctoStatsPage do
+    with true <- function_exported?(Ecto.Repo, :all_running, 0),
+         [_ | _] = repos <- Ecto.Repo.all_running(),
+         [_ | _] = repos_with_extra <- Enum.filter(repos, &extra_available?/1) do
+      {:ok, repos_with_extra}
+    else
+      false ->
+        {:error, :repos_auto_discovery_not_available}
+
+      [] ->
+        {:error, :no_ecto_repos_available}
+    end
+  end
+
+  defp filter_repos_with_extra(repos) do
+    repos_with_extra = Enum.filter(repos, &extra_available?/1)
+
+    if repos_with_extra == [] do
+      {:error, :no_ecto_repos_available}
+    else
+      {:ok, repos_with_extra}
+    end
+  end
+
+  def maybe_assign_nav_repo(socket, params) do
+    nav_repo = params["repo"]
+
+    if nav_repo && socket.assigns[:nav_repo] != nav_repo do
+      nav_repos = for repo <- socket.assigns.repos, do: Atom.to_string(repo)
+
+      if nav_repo in nav_repos do
+        assign(socket, :nav_repo, nav_repo)
+      else
+        socket
+      end
+    else
+      socket
+    end
   end
 
   @impl true
-  def menu_link(%{repo: nil}, _capabilities) do
+  def menu_link(%{repos: []}, _capabilities) do
     if Code.ensure_loaded?(Ecto.Adapters.SQL) do
-      {:disabled, "Ecto Stats", @disabled_link}
+      {:disabled, @page_title, @disabled_link}
     else
       :skip
     end
   end
 
   @impl true
-  def menu_link(%{repo: repo}, capabilities) do
-    title = "#{repo |> inspect() |> String.replace(".", " ")} Stats"
-    extra = info_module_for(repo)
+  def menu_link(%{repos: :auto_discover}, _caps) do
+    case auto_discover() do
+      {:ok, _repos} ->
+        {:ok, @page_title}
+
+      {:error, :repos_auto_discovery_not_available} ->
+        # Page will display the error.
+        {:ok, @page_title}
 
+      {:error, _} ->
+        {:disabled, @page_title, @disabled_link}
+    end
+  end
+
+  @impl true
+  def menu_link(%{repos: repos}, capabilities) do
     cond do
-      repo not in capabilities.processes ->
+      not Enum.any?(repos, fn repo -> repo in capabilities.processes end) ->
         :skip
 
-      extra && Code.ensure_loaded?(extra) ->
-        {:ok, title}
+      extra_available_for_any?(repos) ->
+        {:ok, @page_title}
 
       true ->
-        {:disabled, title, @disabled_link}
+        {:disabled, @page_title, @disabled_link}
     end
   end
 
+  defp extra_available_for_any?(repos) do
+    Enum.any?(repos, &extra_available?/1)
+  end
+
+  defp extra_available?(repo) do
+    extra = info_module_for(repo)
+    extra && Code.ensure_loaded?(extra)
+  end
+
   defp info_module_for(repo) do
     case repo.__adapter__ do
       Ecto.Adapters.Postgres -> EctoPSQLExtras
       _ -> nil
     end
   end
 
+  @impl true
+  def handle_params(params, _url, socket) do
+    # Here we fix the navigation value for "repo" when only "nav" is changing.
+    # This is because we have two nav bars in this page and only one change at once.
+    socket =
+      if params["nav"] && !params["repo"] && socket.assigns[:nav_repo] do
```

Comment:
Instead, I would perhaps add another configuration to `nav`, which is `extra_params: ["repo"]`, and then make sure that all nav links also include the repo option. WDYT?

Reply:
Cool, I will add that param :+1: